internal/dataplane/sendconfig/sendconfig.go

```diff
@@ -100,96 +85,44 @@ func PerformUpdate(ctx context.Context,
 // -----------------------------------------------------------------------------
 
-type InMemoryClient interface {
-	BaseRootURL() string
-	ReloadDeclarativeRawConfig(ctx context.Context, config io.Reader, checkHash bool, flattenErrors bool) ([]byte, error)
+type KonnectAwareClient interface {
+	IsKonnect() bool
 }
 
-func onUpdateInMemoryMode(
-	ctx context.Context,
-	log logrus.FieldLogger,
-	state *file.Content,
-	client InMemoryClient,
-) error {
-	// Kong will error out if this is set
-	state.Info = nil
-	// Kong errors out if `null`s are present in `config` of plugins
-	deckgen.CleanUpNullsInPluginConfigs(state)
-
-	config, err := json.Marshal(state)
-	if err != nil {
-		return fmt.Errorf("constructing kong configuration: %w", err)
-	}
-
-	log.Debug("sending configuration to Kong Admin API")
-	if _, err := client.ReloadDeclarativeRawConfig(ctx, bytes.NewReader(config), true, false); err != nil {
-		return fmt.Errorf("failed sending declarative config to %s: %w", client.BaseRootURL(), err)
-	}
-
-	return nil
+type StatusClient interface {
+	Status(context.Context) (*kong.Status, error)
 }
 
-func onUpdateDBMode(
+// hasConfigurationChanged verifies whether configuration has changed by comparing old and new config's SHAs.
+// In case the SHAs are equal, it still can return true if a client is considered crashed based on its status.
+func hasConfigurationChanged(
 	ctx context.Context,
-	targetContent *file.Content,
-	client *adminapi.Client,
-	dumpConfig dump.Config,
-	config Config,
-) error {
-	cs, err := currentState(ctx, client, dumpConfig)
-	if err != nil {
-		return fmt.Errorf("failed getting current state for %s: %w", client.BaseRootURL(), err)
-	}
-
-	ts, err := targetState(ctx, targetContent, cs, config.Version, client, dumpConfig)
-	if err != nil {
-		return deckConfigConflictError{err}
+	oldSHA, newSHA []byte,
+	client KonnectAwareClient,
+	statusClient StatusClient,
+	log logrus.FieldLogger,
+) (bool, error) {
+	if !bytes.Equal(oldSHA, newSHA) {
+		return true, nil
 	}
-
-	syncer, err := diff.NewSyncer(diff.SyncerOpts{
-		CurrentState:    cs,
-		TargetState:     ts,
-		KongClient:      client.AdminAPIClient(),
-		SilenceWarnings: true,
-	})
-	if err != nil {
-		return fmt.Errorf("creating a new syncer for %s: %w", client.BaseRootURL(), err)
+	if !hasSHAUpdateAlreadyBeenReported(newSHA) {
+		log.Debugf("sha %s has been reported", hex.EncodeToString(newSHA))
 	}
-
-	_, errs := syncer.Solve(ctx, config.Concurrency, false)
-	if errs != nil {
-		return deckutils.ErrArray{Errors: errs}
+	// In case of Konnect, we skip further steps as it doesn't report its configuration hash.
+	if client.IsKonnect() {
+		return false, nil
 	}
 
-	return nil
-}
-
-func currentState(ctx context.Context, kongClient *adminapi.Client, dumpConfig dump.Config) (*state.KongState, error) {
-	rawState, err := dump.Get(ctx, kongClient.AdminAPIClient(), dumpConfig)
+	hasNoConfiguration, err := kongHasNoConfiguration(ctx, statusClient, log)
```

Comment:
`hasNoConfiguration` is a bit hard to understand the true purpose here. how about adding some comments here?

Reply:
Added a comment explaining it.